pkg/cmd/set_replication.go

```diff
@@ -155,6 +155,18 @@ func (o *FinalizeOptions) SetReplication() error {
 		Schedule: repSource.Spec.Trigger.Schedule,
 		Manual:   lastManualSync,
 	}
+
+	repDest := &volsyncv1alpha1.ReplicationDestination{}
+	destNSName := types.NamespacedName{
+		Namespace: o.RepOpts.Dest.Namespace,
+		Name:      o.destName,
+	}
+	if err := o.RepOpts.Dest.Client.Get(ctx, destNSName, repDest); err != nil {
+		return err
+	}
+
+	oldImage := repDest.Status.LatestImage
```

Comment:
You'll need to check `repDest.Status` for nil before dereferencing
Probably something like:
```go
var oldImage *corev1.TypedLocalObjectReference
if repDest.Status != nil {
  oldImage = repDest.Status.LatestImage
}
```